### ISSUE TITLE
spinning up aws instance suddenly failing with compile_time error for fog

### ISSUE THREAD

**Issue by kjhangiani** [hw-cookbooks/route53#88]
We are using Chef 11.10.4 and up until today were using Opsworks Agent 3421 in AWS.   Tried to spin up a server, which failed, and upon logging into AWS saw the following message:

```
Please update the OpsWorks agent version to the latest (3433).
This new version addresses a problem with downloading a dependency during instance setup. Newly launched instances will fail their setup unless you upgrade. You can select the agent version in the stack settings.
```

So, I changed this, but now all instances fail during running_setup with the following error:

```
NoMethodError
-------------
undefined method `compile_time' for Chef::Resource::ChefGem
 
 
Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/route53/recipes/default.rb:24:in `block in from_file'
/var/lib/aws/opsworks/cache.stage2/cookbooks/route53/recipes/default.rb:22:in `from_file'
/var/lib/aws/opsworks/cache.stage2/cookbooks/soxhub/definitions/update_instance_dns.rb:7:in `block in from_file'
/var/lib/aws/opsworks/cache.stage2/cookbooks/soxhub/recipes/update_instance_dns.rb:11:in `block in from_file'
/var/lib/aws/opsworks/cache.stage2/cookbooks/soxhub/recipes/update_instance_dns.rb:10:in `each'
/var/lib/aws/opsworks/cache.stage2/cookbooks/soxhub/recipes/update_instance_dns.rb:10:in `from_file'
/var/lib/aws/opsworks/cache.stage2/cookbooks/soxhub/recipes/init_soxhub_webserver.rb:9:in `from_file'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/route53/recipes/default.rb:
 
17:  # limitations under the License.
18:  #
19:  
20:  include_recipe 'xml::ruby'
21:  
22:  chef_gem "fog" do
23:    action :install
24>>   compile_time false
25:    version node['route53']['fog_version']
26:  end
27:  
28:  require 'rubygems'
29:  Gem.clear_paths
30:  
 
 
[2016-03-09T03:32:29+00:00] ERROR: Running exception handlers
[2016-03-09T03:32:29+00:00] ERROR: Exception handlers complete
[2016-03-09T03:32:29+00:00] FATAL: Stacktrace dumped to /var/lib/aws/opsworks/cache.stage2/chef-stacktrace.out
[2016-03-09T03:32:29+00:00] ERROR: undefined method `compile_time' for Chef::Resource::ChefGem
[2016-03-09T03:32:29+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

I am posting this issue here because my googling has implied that this is where the error originates, but I am just looking for a way to resolve this.  Nothing else has changed in our cookbooks.

This is our Berksfile in our cookbook:

```
source "https://api.berkshelf.com"

cookbook 'build-essential','2.2.4'
cookbook "aws", ">= 2.2.2"
cookbook "route53", "~> 0.4.0"
cookbook 'nodejs', '~> 2.4.0'
cookbook 'database', '~> 4.0.6'
cookbook 'fail2ban', '~> 2.2.1'
```

And a prior developer has left me this cryptic comment in my default.rb:

```
# Override route53 fog_version to fix Fog::AWS::CredentialFetcher bug 
default['route53']['fog_version'] = "1.29.0"
```

Googling that issue leads me to this ticket, which is what I believe that version tag was for:
https://github.com/hw-cookbooks/route53/issues/69

Unfortunately I'm not familiar enough with this aspect of the cookbooks to be able to debug it further.  I'm currently about to try incrementing route53 to 0.4.2 and removing that version override, but any advice would be greatly appreciated.

**Comment by kjhangiani** [comment]
It looks like the issue was caused by the route53 @ 0.4.2 version that was just released.  I was able to resolve this issue, though if you can provide any clarification on why it occurred before closing this ticket, I'd appreciate it.

To summarize:
route53@0.4.2 w/ fog_version overriden to 1.29.0 failed
route53@0.4.2 w/o any fog_version override failed

Ended up locking route53@0.4.0 and keeping the version override to fix the issue.  Am now using the following in my Berksfile:

```
cookbook 'route53', '0.4.0'
```

**Comment by slyness** [comment]
Fixed in `0.4.4` .. released to supermarket
